File: data/code_pairs/cpp/999.cpp
#include <iostream>
#include <cstdint>
#include <climits>

int testSetHighestBit() {
    int8_t i1 = -1;
    return (i1 == -1) ? 0 : 1;
}

int testClearHighestBit() {
    int8_t i1 = 0;
    return (i1 == 0) ? 0 : 1;
}

int testInvertBits() {
    int8_t i1 = ~int8_t(0);
    return (i1 == -1) ? 0 : 1;
}

int main() {
    if (testSetHighestBit() == 0) {
        std::cout << "Test for setting the highest bit Passed\n";
    } else {
        std::cout << "Test for setting the highest bit Failed\n";
    }

    if (testClearHighestBit() == 0) {
        std::cout << "Test for clearing the highest bit Passed\n";
    } else {
        std::cout << "Test for clearing the highest bit Failed\n";
    }

    if (testInvertBits() == 0) {
        std::cout << "Test for inverting bits Passed\n";
    } else {
        std::cout << "Test for inverting bits Failed\n";
    }

    return 0;
}

File: data/code_pairs/fortran/999.f90
program chkbits
  implicit none

  integer :: testResult

  ! Test setting the highest bit
  call testSetHighestBit(testResult)
  if (testResult == 0) then
    print *, "Test for setting the highest bit Passed"
  else
    print *, "Test for setting the highest bit Failed"
  endif

  ! Test clearing the highest bit
  call testClearHighestBit(testResult)
  if (testResult == 0) then
    print *, "Test for clearing the highest bit Passed"
  else
    print *, "Test for clearing the highest bit Failed"
  endif

  ! Test inverting bits
  call testInvertBits(testResult)
  if (testResult == 0) then
    print *, "Test for inverting bits Passed"
  else
    print *, "Test for inverting bits Failed"
  endif

end program chkbits

subroutine testSetHighestBit(result)
  integer, intent(out) :: result
  integer(kind=1) :: i1
  i1 = -1
  if (i1 == -1) then
    result = 0
  else
    result = 1
  endif
end subroutine

subroutine testClearHighestBit(result)
  integer, intent(out) :: result
  integer(kind=1) :: i1
  i1 = 0
  if (i1 == 0) then
    result = 0
  else
    result = 1
  endif
end subroutine

subroutine testInvertBits(result)
  integer, intent(out) :: result
  integer(kind=1) :: i1
  i1 = not(0)
  if (i1 == -1) then
    result = 0
  else
    result = 1
  endif
end subroutine